go/cmd/dolt/commands/diff.go

```diff
@@ -480,30 +507,48 @@ func printShowCreateTableDiff(ctx context.Context, td diff.TableDelta) errhand.V
 	}
 
 	if fromCreateStmt != toCreateStmt {
-		cli.Println(textdiff.LineDiff(fromCreateStmt, toCreateStmt))
+		if fromCreateStmt == "" && toSch.GetAllCols().Size() > fromSch.GetAllCols().Size() {
```

Comment:
If we go with diff_types of added, modified, removed... this could probably be simplified a bit...

You could use `td.IsAdd() && (filter == FilterAdded || filter == NoFilter)` to identify added tables, `td.Drop()` to identify dropped tables, and just rely on everything else being a table modification (e.g. renames, column adds/deletes, etc). Slightly different than the insert/add/delete mapping you proposed, but it would match up better with the existing Dolt diff_types. 

Also... it might be worth creating a small private struct in this file for the filter. Then you could clean up some of the filter checking logic a bit, make the code read a little smoother, and not have to pass around a string as the filter. For example, I'm imagining usage like: 
> if (td.Add() && filter.includeAdds()) { 

That would give a little more typing/structure and also removes the need to check for `NoFilter` directly in many places.